ulib/FStar.ReflexiveTransitiveClosure.fsti

```diff
@@ -0,0 +1,16 @@
+module FStar.ReflexiveTransitiveClosure
+
+open FStar.Preorder
+
+val closure (#a:Type0) (r:relation a) : preorder a
+
+val closure_step: #a:Type0 -> r:relation a -> x:a -> y:a
+  -> Lemma (requires r x y) (ensures closure r x y)
+
+val closure_inversion: #a:Type0 -> r:relation a -> x:a -> y:a
+  -> Lemma (requires closure r x y)
+          (ensures  x == y \/ r x y \/ (exists z. closure r x z /\ closure r z y))
+
+val stable_on_closure: #a:Type0 -> r:relation a -> p:(a -> Type0)
+  -> p_stable_on_r: (squash (forall x y. p x /\ r x y ==> p y))
```

Comment:
A pattern on this quantifier is less important, since it will typically appear in the goal.
However, to be on the safe side, how about `p y; r x y`, in support of backwards reasoning.

Reply:
I added `p y; r x y` as a pattern.